### ISSUE TITLE
Each keypress has cumulative delay when working on a slow network share

### ISSUE THREAD

**Issue by chtenb** [helix-editor/helix#12231]
**Reproduction**
Open a file on a slow network share and notice that each keypress has a delay (in my case about 2 seconds). Even if I press 5 keys in quick succession, they still take 10 seconds in total to complete. This is not only for selection movements, but also keystrokes in command mode.

If I open a scratch buffer with `:new`, this slowness goes away immediately.
If I open a file on my local disk (which is normally fast) helix remains slow, until I change the `cwd` to be outside the network share as well.
Setting the language to `text` with `:set-language` doesn't change the responsiveness.

*Operating system*: windows 11
*Logs*: show nothing out of interest
*Installation method*: compile from source
*Config*: I passed -c /dev/null to ensure default config

Let me know if I can do any other tests to troubleshoot this issue.

**In summary**
If either the current file or the cwd is on a slow network share, each keypress has a cumulative delay

**Comment by justinlovinger** [comment]
Try disabling `path-completion`.
